packages/flutter/lib/src/painting/notched_shapes.dart

```diff
@@ -111,3 +118,44 @@ class CircularNotchedRectangle implements NotchedShape {
       ..close();
   }
 }
+
+/// A [NotchedShape] created from [ShapeBorder]s.
+///
+/// Two shapes can be provided. The [host] is the shape of the widget that
+/// uses the [NotchedShape] (typically a [BottomAppBar]). The [guest] is
+/// subtracted from the [host] to create the notch (typically to make room
+/// for a [FloatingActionButton]).
+class AutomaticNotchedShape extends NotchedShape {
```

Comment:
bonus points for adding a code example showing this in use (additional bonus points for including an image that shows the result).

Reply:
I'm going to pass on this for now... I wasn't quite able to figure out what the right way to demonstrate this was.